flink-runtime/src/test/java/org/apache/flink/runtime/externalresource/ExternalResourceUtilsTest.java

```diff
@@ -0,0 +1,247 @@
+		final Configuration config = new Configuration();
+
+		config.set(ExternalResourceOptions.EXTERNAL_RESOURCE_LIST, RESOURCE_LIST);
+		config.setLong(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_AMOUNT_SUFFIX), RESOURCE_AMOUNT_1);
+		config.setLong(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_2, ExternalResourceOptions.EXTERNAL_RESOURCE_AMOUNT_SUFFIX), RESOURCE_AMOUNT_2);
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, SUFFIX), RESOURCE_CONFIG_KEY_1);
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_2, SUFFIX), RESOURCE_CONFIG_KEY_1);
+
+		final Map<String, Long> configMap = ExternalResourceUtils.getExternalResources(config, SUFFIX);
+
+		// Only one of the config key would be kept.
+		assertThat(configMap.size(), is(1));
+		assertTrue(configMap.containsKey(RESOURCE_CONFIG_KEY_1));
+	}
+
+	@Test
+	public void testGetExternalResourcesWithMultipleExternalResource() {
+		final Configuration config = new Configuration();
+
+		config.set(ExternalResourceOptions.EXTERNAL_RESOURCE_LIST, RESOURCE_LIST);
+		config.setLong(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_AMOUNT_SUFFIX), RESOURCE_AMOUNT_1);
+		config.setLong(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_2, ExternalResourceOptions.EXTERNAL_RESOURCE_AMOUNT_SUFFIX), RESOURCE_AMOUNT_2);
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, SUFFIX), RESOURCE_CONFIG_KEY_1);
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_2, SUFFIX), RESOURCE_CONFIG_KEY_2);
+
+		final Map<String, Long> configMap = ExternalResourceUtils.getExternalResources(config, SUFFIX);
+
+		assertThat(configMap.size(), is(2));
+		assertTrue(configMap.containsKey(RESOURCE_CONFIG_KEY_1));
+		assertTrue(configMap.containsKey(RESOURCE_CONFIG_KEY_2));
+		assertThat(configMap.get(RESOURCE_CONFIG_KEY_1), is(RESOURCE_AMOUNT_1));
+		assertThat(configMap.get(RESOURCE_CONFIG_KEY_2), is(RESOURCE_AMOUNT_2));
+	}
+
+	@Test
+	public void testConstructExternalResourceDriversFromConfig() {
+		final Configuration config = new Configuration();
+		final String driverFactoryClassName = TestingExternalResourceDriverFactory.class.getName();
+		final Map<Class<?>, Iterator<?>> plugins = new HashMap<>();
+		plugins.put(ExternalResourceDriverFactory.class, IteratorUtils.singletonIterator(new TestingExternalResourceDriverFactory()));
+		final PluginManager testingPluginManagerPluginManager = new TestingPluginManager(plugins);
+
+		config.set(ExternalResourceOptions.EXTERNAL_RESOURCE_LIST, Collections.singletonList(RESOURCE_NAME_1));
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX), driverFactoryClassName);
```

Comment:
Here we could introduce a `ExternalResourceOptions.getExternalResourceDriverFactoryConfigOption(RESOURCE_NAME_1)`.